### DISCUSSION TITLE
error: PAL failed setting personality failed: Operation not permitted (EPERM)

### DISCUSSION THREAD

**Discussion by JINGBO314** [gramineproject/gramine#1729] [General]
How can I solve this problem? This problem occurs when I run gramine in ubuntu20.04 running in docker.

**Comment by kailun-qin** [comment]
@duanjingbo314: Were you trying to run `gramine-direct` within your docker container? If so, I think you'll need to pass Gramine's [custom seccomp profile](https://github.com/gramineproject/gramine/blob/master/scripts/docker_seccomp_aug_2022.json). Pls see more details here: https://gramine.readthedocs.io/en/latest/installation.html#use-gramine-docker-image.

**Reply by JINGBO314** [reply]
thank you for your reply！！！I only used gramine-direct, and the above error also occurred under this condition, because I wanted to install gramine directly in my existing docker container and then execute it, just like running it in a normal Ubuntu system;I will try the official method later.
